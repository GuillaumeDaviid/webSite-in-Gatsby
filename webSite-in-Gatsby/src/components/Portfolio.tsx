import React, { useState } from 'react'
import './styles/Portfolio.scss'
import { StaticImage } from 'gatsby-plugin-image'
import PortfolioItems from './PortfolioItems'

export default function Portfolio() {
    const [selected, setSelected] = useState('all');

    function handleClick(category:string) {
        setSelected(category)
    }

    return(
        <section id="portfolio">
        <div className="Portfolio">
       
            <h2 className="Protfolio_title" data-testid="title">MON PORTFOLIO</h2>

            <div className="Portfolio_separator" data-testid="separator"></div>
            

            <ul className="Portfolio_btn">
                <li className={`${selected === "all" ? `selected` : `notSelected`}`} onClick={() => handleClick("all")} data-testid="all">ALL</li>
                <li className={`${selected === "front" ? `selected` : `notSelected`}`} onClick={() => handleClick("front")} data-testid="front">FRONT END</li>
                <li className={`${selected === "full" ? `selected` : `notSelected`}`} onClick={() => handleClick("full")} data-testid="full">FULL STACK</li>
                <li className={`${selected === "mobile" ? `selected` : `notSelected`}`} onClick={() => handleClick("mobile")} data-testid="mobile">MOBILE</li>
            </ul>
            <div className="Portfolio_items">

                <PortfolioItems selected={selected} image={<StaticImage alt="Smartphone responsive" src='../images/netflix.png' />} link='https://github.com/GuillaumeDaviid/Netflix-Clone' title='Netflix Clone' full={false}/>

                <PortfolioItems selected={selected} image={<StaticImage alt="Smartphone responsive" src='../images/checkers.png' />} link='https://guillaumedaviid.github.io/Checkers-Game/' title='Jeu de Dame' full={false}/>

                <PortfolioItems selected={selected} image={<StaticImage alt="Smartphone responsive" src='../images/01academie.png' />} link='https://github.com/GuillaumeDaviid/01-Academie-Version-React' title='01 ACADEMIE' full/>

                <PortfolioItems selected={selected} image={<StaticImage alt="Smartphone responsive" src='../images/01courses.png' />} link='https://github.com/GuillaumeDaviid/01-Academie-Version-React' title='01 ACADEMIE' full/>

                <PortfolioItems selected={selected} image={<StaticImage alt="Smartphone responsive" src='../images/01home.png' />} link='https://github.com/GuillaumeDaviid/01-Academie-Version-React' title='01 ACADEMIE' full/>

              
            </div>
        </div>
        </section>
    )
}