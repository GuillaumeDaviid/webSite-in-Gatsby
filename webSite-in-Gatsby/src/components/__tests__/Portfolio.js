import React from "react"
import renderer from "react-test-renderer"
import Portfolio from "../Portfolio.tsx"
import { render, screen } from "@testing-library/react"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Portfolio />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Portfolio />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a separator element", () => {
    render(<Portfolio />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have a All element", () => {
    render(<Portfolio />)

    const all = screen.getByTestId('all');
    expect(all).toBeTruthy();
  })

  it("should have a front element", () => {
    render(<Portfolio />)

    const front = screen.getByTestId('front');
    expect(front).toBeTruthy();
  })

  it("should have a full element", () => {
    render(<Portfolio />)

    const full = screen.getByTestId('full');
    expect(full).toBeTruthy();
  })

  it("should have a mobile element", () => {
    render(<Portfolio />)

    const mobile = screen.getByTestId('mobile');
    expect(mobile).toBeTruthy();
  })

  it("should have 5 portfolio elements", () => {
    render(<Portfolio />)

    const portfolio = screen.getAllByTestId('portfolio');
    expect(portfolio).toHaveLength(5);
  })

  it("should have 5 links in portfolio element", () => {
    render(<Portfolio />)

    const portfolio = screen.getAllByTestId('link');
    expect(portfolio).toHaveLength(5);
  })
})