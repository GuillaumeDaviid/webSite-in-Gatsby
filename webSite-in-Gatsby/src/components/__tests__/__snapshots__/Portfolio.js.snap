// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<section
  id="portfolio"
>
  <div
    className="Portfolio"
  >
    <h2
      className="Protfolio_title"
      data-testid="title"
    >
      MON PORTFOLIO
    </h2>
    <div
      className="Portfolio_separator"
      data-testid="separator"
    />
    <ul
      className="Portfolio_btn"
    >
      <li
        className="selected"
        data-testid="all"
        onClick={[Function]}
      >
        All
      </li>
      <li
        className="notSelected"
        data-testid="front"
        onClick={[Function]}
      >
        Front end
      </li>
      <li
        className="notSelected"
        data-testid="full"
        onClick={[Function]}
      >
        Full stack
      </li>
      <li
        className="notSelected"
        data-testid="mobile"
        onClick={[Function]}
      >
        Mobile
      </li>
    </ul>
    <div
      className="Portfolio_items"
    >
      <div
        className="Portfolio_items-elt visible"
        data-testid="portfolio"
      >
        <div
          className="Portfolio_items-content"
        >
          <h3>
            Netflix Clone
          </h3>
          <a
            data-testid="link"
            href="https://github.com/GuillaumeDaviid/Netflix-Clone"
            target="_blank"
          >
            Link
          </a>
        </div>
      </div>
      <div
        className="Portfolio_items-elt visible"
        data-testid="portfolio"
      >
        <div
          className="Portfolio_items-content"
        >
          <h3>
            Jeu de Dame
          </h3>
          <a
            data-testid="link"
            href="https://guillaumedaviid.github.io/Checkers-Game/"
            target="_blank"
          >
            Link
          </a>
        </div>
      </div>
      <div
        className="Portfolio_items-elt visible"
        data-testid="portfolio"
      >
        <div
          className="Portfolio_items-content"
        >
          <h3>
            01 ACADEMIE
          </h3>
          <a
            data-testid="link"
            href="https://www.01academie.com/"
            target="_blank"
          >
            Link
          </a>
        </div>
      </div>
      <div
        className="Portfolio_items-elt visible"
        data-testid="portfolio"
      >
        <div
          className="Portfolio_items-content"
        >
          <h3>
            01 ACADEMIE
          </h3>
          <a
            data-testid="link"
            href="https://www.01academie.com/"
            target="_blank"
          >
            Link
          </a>
        </div>
      </div>
      <div
        className="Portfolio_items-elt visible"
        data-testid="portfolio"
      >
        <div
          className="Portfolio_items-content"
        >
          <h3>
            01 ACADEMIE
          </h3>
          <a
            data-testid="link"
            href="https://www.01academie.com/"
            target="_blank"
          >
            Link
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
